core_lib/src/interface/editor.cpp

```diff
@@ -494,6 +489,21 @@ void Editor::updateAutoSaveCounter()
     }
 }
 
+void Editor::updateActiveFrames(int frame)
+{
+    int beginFrame = std::max(frame - 3, 1);
```

Comment:
Setting this to frame during playback might help speed things up a little bit since you only need to load frames in front of the current frame. On a similar note, have you considered loading the frames at the beginning of a range when you are close to the end of the range so that looping is smooth?

Reply:
Currently, we load frames in `canvaspainter.cpp` line 208, just before the frame gets drawn. ActiveFramePool only does unloading. A frame in ActiveFramePool doesn't mean it will get loaded.

I'm still thinking how to do it in a better way.

Reply:
Ah I see. Well how about loading the frames in ActiveFramePool::put? That class already handles the unloading anyway.

Reply:
Right, we can do that. And probably it's a good idea to preload the frames.